### PR TITLE
CLOUDP-229283: disable ownership detection

### PR DESCRIPTION
**Pull request by s-urbaniak**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -225,6 +232,31 @@ func main() {
 		os.Exit(1)
 	}
 
+	err = mgr.Add(manager.RunnableFunc(func(ctx context.Context) error {
+		setupLog.Info(SUBOBJECT_DELETION_PROTECTION_MESSAGE)
+
+		akoPod := &corev1.Pod{}
+		err = mgr.GetClient().Get(ctx, client.ObjectKey{
```
Comment (s-urbaniak) [inline cmd/manager/main.go:239]: recheck if we need RBAC for this

**Review by igor-karpukhin — APPROVED** [review]
Great fix! 👍

**Review by josvazg — APPROVED** [review]
LGTM!

Be aware though, that this is missing the subobject checks. Those are not called `isOwner`, but instead `can...Reconcile`, eg. `canUditingReconcile`.

BTW, the `IsOwner` is used on top level objects that we are pretty sure will remove the check for good, with no replacement. Where as the `can...Reconcile` will need to be revisited with a proper alternative.

An useful regexp to find the affected code is `can.*Reconcile` which renders me 36 results in 12 files. I would vote to do that in a separate PR.

These are

**Comment by s-urbaniak** [comment]
@josvazg if there is something missing, we should definitely add it to this PR:

> Be aware though, that this is missing the subobject checks. Those are not called isOwner, but instead can...Reconcile, eg. canUditingReconcile.

This definitely should be covered by setting `SubObjectDeletionProtection: false` at the reconciler struct level. Please advise what else we should do here.

**Comment by josvazg** [comment]
> @josvazg if there is something missing, we should definitely add it to this PR:
> 
> > Be aware though, that this is missing the subobject checks. Those are not called isOwner, but instead can...Reconcile, eg. canUditingReconcile.
> 
> This definitely should be covered by setting `SubObjectDeletionProtection: false` at the reconciler struct level. Please advise what else we should do here.

Yes, they are also disabled already, but maybe it would be good to mark the code as disabled explicitly?
We could skip and just go for a later removal. Not a big deal.

**Comment by josvazg** [comment]
> We could skip and just go for a later removal. Not a big deal.

Lets do that and move on.

File: cmd/manager/main.go (hunk 1 above)
Comment (s-urbaniak) [reply cmd/manager/main.go]: @igor-karpukhin @josvazg: we indeed do need additional RBAC here which we don't ship today.

Comment (s-urbaniak) [reply cmd/manager/main.go]: Removing emitting the event on the AKO pod for now. As we found, the inclusion of additional RBAC permissions is too intrusive for this release.

**Pull request merged**